File: golang/src/entidades/db/carro.go
package db

type Carro struct {
	Id   uint   `gorm:"column:id;primaryKey" json:"id" form:"id"`
	Nome string `gorm:"column:nome" json:"nome" form:"nome"`
	Cor  string `gorm:"column:cor" json:"cor" form:"cor"`
	Ano  int    `gorm:"column:ano" json:"ano" form:"ano"`

	DonoId uint   `gorm:"column:pessoa_id" json:"pessoa" form:"pessoa"`
	Dono   Pessoa `gorm:"foreignKey:DonoId" json:"dono"`
}

func (Carro) TableName() string {
	return "carro"
}

File: nodejs/src/entidades/db/carro.ts
import { DataTypes } from 'sequelize';
import { db } from '../../config/db';

import { Pessoa } from './pessoa';

export const Carro = db.define('Carro', {
  id: {
    primaryKey: true,
    type: DataTypes.INTEGER,
    autoIncrement: true
  },
  nome: {
    type: DataTypes.STRING(255)
  },
  cor: {
    type: DataTypes.STRING(45)
  },
  ano: {
    type: DataTypes.INTEGER,
    defaultValue: 0
  },
  pessoa_id: {
    type: DataTypes.INTEGER,
    references: {
      model: Pessoa,
      key: 'id',
    }
  }
}, {
  tableName: 'carro',
  timestamps: false,
});
// Carro.beforeAssociate = (models, others) => {
//   console.log('Carro.beforeAssociate', models, others);
// };
